sphinx_gallery/gen_gallery.py

```diff
@@ -180,11 +182,22 @@ def generate_gallery_rst(app):
 
         computation_times += this_computation_times
 
+        key = None
+        if gallery_conf['examples_dirs_order'] is not None:
+            base_examples_dir = os.path.basename(examples_dir)
```

Comment:
This will be very prone to failure. This is because `examples_dir` is an absolute path, but with `os.path.basename` you just get the last folder name. Nothing guaranties you that it will match with the user folder path. I think we should move this code block inside `_prepare_sphx_glr_dirs`, that one has the user's original paths, and we can make it return the absolute paths to work on + the sorting key function